### PR TITLE
Migration: drop too long repo topics

### LINKED ISSUE #14108: DB topic.name uses varying(25) which makes mirrors of repos with long topics fail
- Gitea version: 1.13.0 (also fails on latest dev)
- Git version: 2.26.2
- Operating system:
  - `docker-compose` with the image `gitea/gitea:1`
- Database:
  - [x] PostgreSQL

## Description

1. Create new migration
2. Select GitHub
3. Clone `https://github.com/survivejs/webpack-book` as a **mirror**
4. Will fail with: 

```
Migrate repository from https://github.com/survivejs/webpack-book failed: pq: value too long for type character varying(25)
```

Can be solved by using `varying(255)` on `topic.name`. Is there a reason it's limited to 25?

### PR DESCRIPTION
**Pull request by 6543**
close #14108

follow up (not backportable): #14150

### CODE REVIEW

File: modules/migrations/gitea_uploader.go [hunk 1]
```diff
@@ -154,6 +154,15 @@ func (g *GiteaLocalUploader) Close() {
 
 // CreateTopics creates topics
 func (g *GiteaLocalUploader) CreateTopics(topics ...string) error {
+	// ignore topics to long for the db
+	c := 0
+	for i := range topics {
+		if len(topics[i]) <= 25 {
```
Comment (zeripath) [inline modules/migrations/gitea_uploader.go]: Once #14150 is merged this will be 50

Comment (techknowlogick) [reply modules/migrations/gitea_uploader.go]:
```suggestion
		if len(topics[i]) <= 50 {
```

**Review by zeripath — APPROVED** [review]
I think a more programmatic way of detecting these limits would be better but this will suffice

**Comment by 6543** [comment]
@techknowlogick changed to 50

**Comment by 6543** [comment]
Backport: #14155

**Review by lunny — APPROVED** [review]

**Pull request merged**